src/_P121_HMC5883L.ino

```diff
@@ -127,7 +127,7 @@ boolean Plugin_121(uint8_t function, struct EventStruct *event, String &string)
 
       float heading = atan2(s_event.magnetic.y, s_event.magnetic.x);
 
-      const float decl = PCONFIG(0);
+      const float decl = PCONFIG_FLOAT(0);
       addLog(LOG_LEVEL_DEBUG, "Declination Degree:" + String(decl));
```

Comment:
Just FYI, the default number of decimals for the `String(float, int nrdecimals = 2)` constructor is 2 decimals.

Also you need to use the F() macro for strings or else you end up using quite a lot of resources on the ESP.
And please enclose those debug log level things in wrappers like these:
```c++
#ifndef BUILD_NO_DEBUG
if (logLevelActiveFor(LOG_LEVEL_DEBUG) {
  ...
}
#endif
```